addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -4454,10 +4463,154 @@ registry.ImageOptimize = ImageHandlerOption.extend({
+     * @returns Returns a promise
+     */
+    async _applyShape(shape, previewMode) {
+        const img = this._getImg();
+        if (!shape) {
+            if (!img.dataset.shape) {
+                return;
+            }
+            const svgUrl = img.getAttribute('src');
+            const svgText = await (await fetch(svgUrl)).text();
+            const SVGParser = new DOMParser();
+            const svg = SVGParser.parseFromString(svgText, 'image/svg+xml').documentElement;
+            const svgImage = svg.querySelector('image');
+            img.src = svgImage.getAttribute('xlink:href');
+            img.mimetype = 'image/jpeg';
+            delete img.dataset.shape;
+            return;
+        }
+        const [module, fileName] = shape.split('/');
+        const clipShapeURL = `/${module}/static/image_shapes/${fileName}.svg`;
+        const clipShape = await (await fetch(clipShapeURL)).text();
+        const SVGParser = new DOMParser();
+        const svg = SVGParser.parseFromString(clipShape, 'image/svg+xml').documentElement;
+
+        const base64 = await applyModifications(img);
+
+        const svgImage = svg.querySelector('image');
+        svgImage.setAttribute('xlink:href', base64);
+        svg.setAttribute('width', img.naturalWidth);
+        svg.setAttribute('height', img.naturalHeight);
+        img.dataset.shape = shape;
+        await this._writeSvg(svg.outerHTML, fileName);
+        return this._applyColor(img.dataset.shapeColor, true, previewMode);
+    },
+
+    /**
+     * Applies the correct color to the shape by looking for the default palette's hexcode inside the OG svg files
+     * @param {string} color HEX Code for the color of the background shape
+     * @param {bool} newShape wether it's a new shape or not so we know to look for default colors or previous ones
+     * @param {string} previewMode state of the editor to know which colors to look for
+     */
+    async _applyColor(color, newShape, previewMode) {
+        const img = this._getImg();
+        if (!img.dataset.shape) {
+            return;
+        }
+
+        const fileName = img.dataset.shape.split('/')[1];
+        const clipShapeURL = img.src;
+        let clipShape = await (await fetch(clipShapeURL)).text();
+        let newColor = color && normalizeColor(color);
+        if (newShape) {
+            const DEFAULT_PALETTE = weUtils.DEFAULT_PALETTE;
+            const combinedColorsRegex = new RegExp(Object.values(DEFAULT_PALETTE).join('|'), 'gi');
+            if (previewMode === 'reset' && img.dataset.shapeColor) {
+                clipShape = clipShape.replace(combinedColorsRegex, img.dataset.shapeColor);
+            } else {
+                clipShape = clipShape.replace(combinedColorsRegex, match => {
+                    const colorId = Object.keys(DEFAULT_PALETTE).find(key => DEFAULT_PALETTE[key] === match.toUpperCase());
+                    return newColor = weUtils.getCSSVariableValue('o-color-' + colorId);
+                });
```

Comment:
It looks like this part supports multiple colors, but none of the rest of this function seems to support that. Can you add support for multiple colors? You can try that by adding  an extra element to the background of one shape that uses a different color from the default palette